stack-graphs/src/cycles.rs

```diff
@@ -84,7 +84,7 @@ impl HasPathKey for PartialPath {
     }
 }
 
-const MAX_SIMILAR_PATH_COUNT: usize = 12;
+const MAX_SIMILAR_PATH_COUNT: usize = 7;
```

Comment:
:rofl: Cycle detection by magic number

Reply:
I upped it a lot for a single test. I decided to lower it again and skip that one test for now, so we don't end up with such a high number in `main`. Once we've sorted this out properly, those tests can be enabled again.